app/main/views/frameworks.py

```diff
@@ -54,18 +50,24 @@ def framework_dashboard():
 def framework_services():
     template_data = main.config['BASE_TEMPLATE_DATA']
 
-    try:
-        drafts = data_api_client.find_draft_services(
-            current_user.supplier_id,
-            framework='g-cloud-7'
-        )['services']
+    drafts, complete_drafts = get_drafts(data_api_client, current_user.supplier_id, 'g-cloud-7')
 
-    except APIError as e:
-        abort(e.status_code)
+    for draft in drafts:
+        draft['priceString'] = format_service_price(draft)
+        content = new_service_content.get_builder().filter(draft)
+        sections = get_service_attributes(draft, content)
+
+        draft['unanswered_questions'] = len([
+            q.label
+            for s in sections
+            for q in s['rows']
+            if q.answer_required
+        ])
```

Comment:
This feels like it should be a function in service helpers. It's used again [here](https://github.com/alphagov/digitalmarketplace-supplier-frontend/pull/169/files#diff-32bf67afda65ad26faef3bf3efd522b1R364)